src/trafficcontroller/app/traffic_controller.go

```diff
@@ -0,0 +1,203 @@
+	return &trafficController{
+		conf:                 c,
+		logFilePath:          path,
+		disableAccessControl: disableAccessControl,
+	}
+}
+
+func (t *trafficController) Start() {
+	SetInsecureSkipVerify(t.conf.SkipCertVerify)
+
+	log.Print("Startup: Setting up the loggregator traffic controller")
+
+	batcher, err := t.initializeMetrics("LoggregatorTrafficController", net.JoinHostPort(t.conf.MetronHost, strconv.Itoa(t.conf.MetronPort)))
+	if err != nil {
+		log.Printf("Error initializing dropsonde: %s", err)
+	}
+
+	monitorInterval := time.Duration(t.conf.MonitorIntervalSeconds) * time.Second
+	uptimeMonitor := monitor.NewUptime(monitorInterval)
+	go uptimeMonitor.Start()
+	defer uptimeMonitor.Stop()
+
+	openFileMonitor := monitor.NewLinuxFD(monitorInterval)
+	go openFileMonitor.Start()
+	defer openFileMonitor.Stop()
+
+	etcdAdapter := t.defaultStoreAdapterProvider(t.conf)
+	err = etcdAdapter.Connect()
+	if err != nil {
+		panic(fmt.Errorf("Unable to connect to ETCD: %s", err))
+	}
+
+	logAuthorizer := auth.NewLogAccessAuthorizer(t.disableAccessControl, t.conf.ApiHost)
+
+	uaaClient := auth.NewUaaClient(t.conf.UaaHost, t.conf.UaaClient, t.conf.UaaClientSecret)
+	adminAuthorizer := auth.NewAdminAccessAuthorizer(t.disableAccessControl, &uaaClient)
+
+	finder := dopplerservice.NewFinder(etcdAdapter, int(t.conf.DopplerPort), int(t.conf.GRPC.Port), []string{"ws"}, "")
+	finder.Start()
+
+	var accessMiddleware func(auth.HttpHandler) *auth.AccessHandler
+	if t.conf.SecurityEventLog != "" {
+		accessLog, err := os.OpenFile(t.conf.SecurityEventLog, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+		if err != nil {
+			panic(fmt.Errorf("Unable to open access log: %s", err))
+		}
+		defer func() {
+			accessLog.Sync()
+			accessLog.Close()
+		}()
+		accessLogger := auth.NewAccessLogger(accessLog)
+		accessMiddleware = auth.Access(accessLogger, t.conf.IP, t.conf.OutgoingDropsondePort)
+	}
+
+	creds, err := plumbing.NewCredentials(
+		t.conf.GRPC.CertFile,
+		t.conf.GRPC.KeyFile,
+		t.conf.GRPC.CAFile,
+		"doppler",
+	)
+	if err != nil {
+		log.Fatalf("Could not use GRPC creds for server: %s", err)
+	}
+
+	pool := plumbing.NewPool(20, grpc.WithTransportCredentials(creds))
+	grpcConnector := plumbing.NewGRPCConnector(1000, pool, finder, batcher)
+
+	dopplerHandler := http.Handler(proxy.NewDopplerProxy(logAuthorizer, adminAuthorizer, grpcConnector, "doppler."+t.conf.SystemDomain, 15*time.Second))
+	if accessMiddleware != nil {
+		dopplerHandler = accessMiddleware(dopplerHandler)
+	}
+	t.startOutgoingProxy(fmt.Sprintf(":%d", t.conf.OutgoingDropsondePort), dopplerHandler)
+
+	killChan := signalmanager.RegisterKillSignalChannel()
+	dumpChan := signalmanager.RegisterGoRoutineDumpSignalChannel()
+
+	// We start the profiler last so that we can definitively claim that we're ready for
+	// connections by the time we're listening on the PPROFPort.
+	p := profiler.New(t.conf.PPROFPort)
```

Comment:
On other components we put the pprof binding in the main.

https://github.com/cloudfoundry-incubator/scalable-syslog/blob/0fa3b79fa50fc1b12658e8ec0f9e19829205a512/adapter/main.go#L49
https://github.com/cloudfoundry-incubator/scalable-syslog/blob/0fa3b79fa50fc1b12658e8ec0f9e19829205a512/scheduler/main.go#L53

I'm not sure all of this belongs in the app.